src/CommunityToolkit.Maui.MediaElement/Views/MediaManager.android.cs

```diff
@@ -48,9 +65,129 @@ public partial class MediaManager : Java.Lang.Object, IPlayer.IListener
+	{
+		var status = await Permissions.CheckStatusAsync<AndroidMediaPermissions>().WaitAsync(cancellationToken).ConfigureAwait(false);
+		if (status == PermissionStatus.Granted)
+		{
+			return;
+		}
+		else if (status == PermissionStatus.Denied && Build.VERSION.SdkInt > BuildVersionCodes.O)
+		{
+			await Shell.Current.DisplayAlert("Permission Required", "Notification permission is required to show media controls.", "Ok");
+		}
+		await Permissions.RequestAsync<AndroidMediaPermissions>().WaitAsync(cancellationToken).ConfigureAwait(false);
+	}
+
+	async Task StartService(CancellationToken cancellationToken = default)
+	{
+		await checkPermissions;
+		var bitmap = await GetBitmapFromUrl(MediaElement.MetaDataArtworkUrl, Platform.AppContext.Resources, cancellationToken);
+		var mediaMetaData = new MediaMetadataCompat.Builder();
+		mediaMetaData.PutString(MediaMetadataCompat.MetadataKeyArtist, MediaElement.MetaDataArtist);
+		mediaMetaData.PutString(MediaMetadataCompat.MetadataKeyTitle, MediaElement.MetaDataTitle);
+		mediaMetaData.PutBitmap(MediaMetadataCompat.MetadataKeyAlbumArt, bitmap);
+		mediaMetaData.PutLong(MediaMetadataCompat.MetadataKeyDuration, Player?.Duration ?? 0);
+		mediaMetaData.Build();
+
+		mediaSession?.SetMetadata(mediaMetaData.Build());
+		var intent = new Intent(Android.App.Application.Context, typeof(MediaControlsService));
+		if (Build.VERSION.SdkInt >= BuildVersionCodes.O)
+		{
+			intent.PutExtra("title", MediaElement.MetaDataTitle);
+			intent.PutExtra("artist", MediaElement.MetaDataArtist);
+			intent.PutExtra("albumArtUri", MediaElement.MetaDataArtworkUrl);
+			intent.PutExtra("position", ((long)MediaElement.Position.TotalSeconds));
+			intent.PutExtra("currentTime", SystemClock.ElapsedRealtime());
+			intent.PutExtra("duration", ((long)MediaElement.Duration.TotalSeconds));
+		}
+
+		intent.PutExtra("token", mediaSession?.SessionToken);
+		if (Build.VERSION.SdkInt >= BuildVersionCodes.O)
+		{
+			Android.App.Application.Context.StartForegroundService(intent);
+		}
+		else
+		{
+			Android.App.Application.Context.StartService(intent);
+		}
+	}
+
+	void BroadcastUpdate(string action)
+	{
+		if (Build.VERSION.SdkInt >= BuildVersionCodes.Tiramisu)
+		{
+			Logger?.LogError("{LocalBroadcastManager} not supported on Android 13 and above.", typeof(LocalBroadcastManager));
+			return;
+		}
+		Intent intent = new(MediaControlsService.ACTION_UPDATE_UI);
+		intent.PutExtra("ACTION", action);
+		LocalBroadcastManager.GetInstance(Platform.AppContext).SendBroadcast(intent);
+	}
+
+	public static async Task<Bitmap?> GetBitmapFromUrl(string? url, Resources? resources, CancellationToken cancellationToken = default)
+	{
+		var temp = BitmapFactory.DecodeResource(resources, Resource.Drawable.exo_ic_default_album_image);
+		if (client is null)
+		{
+			return temp;
+		}
+
+		try
+		{
+			var response = await client.GetAsync(url, cancellationToken).ConfigureAwait(false);
```

Comment:
no matter what I get the beatles album art, is that correct or it's a placeholder for tests?

Reply:
Its a placeholder. I need a source for artwork that is open source so we can show a demonstration image.

Reply:
I have been unable to find a url with a suitable image that is open source for the sample app. I have spent a few hours looking and I hope someone can help out with this. 

Using something like the dotnet bot would require me to add additional support for file and/or Resources into the PR which I have been told directly is not needed for an initial support. I was advised not to overdo it. If that is the only option I will add the support. But I need a direct answer to that from someone on the team that is ok to go and add support for this. That will require device specific code for each type and will also require tests. That could delay the PR significantly.

Reply:
Used ChatGPT to create an image based on dotnet bot! 

![DALL·E 2024-04-24 15 50 32 - A small, futuristic purple robot wearing headphones, standing outdoors on a sunny autumn day  The robot is compact and adorable, with glowing eyes and](https://github.com/CommunityToolkit/Maui/assets/4167863/269723c4-4ded-443b-a8a4-46c2838e4752)